subsys/net/lib/sockets/Kconfig

```diff
@@ -113,6 +113,15 @@ config NET_SOCKETS_OFFLOAD
 	  See NET_OFFLOAD for a more deeply integrated approach which offloads
 	  from the net_context() API within the Zephyr IP stack.
 
+config NET_SOCKETS_OFFLOAD_TLS
+	bool "Offload TLS socket calls to the modem"
```

Comment:
I think we do not need to mention `modem` here, as the offloading could be used in other type of devices too.